staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -457,35 +441,28 @@ func (t *tracker) Delete(gvr schema.GroupVersionResource, ns, name string) error
 	t.lock.Lock()
 	defer t.lock.Unlock()
 
-	found := false
-
-	for i, existingObj := range t.objects[gvr] {
-		objMeta, err := meta.Accessor(existingObj)
-		if err != nil {
-			return err
-		}
-		if objMeta.GetNamespace() == ns && objMeta.GetName() == name {
-			obj := t.objects[gvr][i]
-			t.objects[gvr] = append(t.objects[gvr][:i], t.objects[gvr][i+1:]...)
-			for _, w := range t.getWatches(gvr, ns) {
-				w.Delete(obj)
-			}
-			found = true
-			break
-		}
+	objs, ok := t.objects[gvr]
+	if !ok {
+		return errors.NewNotFound(gvr.GroupResource(), name)
 	}
 
-	if found {
-		return nil
+	namespacedName := types.NamespacedName{Namespace: ns, Name: name}
+	obj, ok := objs[namespacedName]
+	if !ok {
+		return errors.NewNotFound(gvr.GroupResource(), name)
 	}
 
-	return errors.NewNotFound(gvr.GroupResource(), name)
+	delete(objs, namespacedName)
+	for _, w := range t.getWatches(gvr, ns) {
+		w.Delete(obj)
+	}
+	return nil
 }
 
 // filterByNamespace returns all objects in the collection that
 // match provided namespace. Empty namespace matches
 // non-namespaced objects.
-func filterByNamespace(objs []runtime.Object, ns string) ([]runtime.Object, error) {
+func filterByNamespace(objs map[types.NamespacedName]runtime.Object, ns string) ([]runtime.Object, error) {
```

Comment:
doesn't this make the order returned by List non-deterministic?

Reply:
You are right, the order is non-deterministic, is it a concern? I think even the result of the real clientset is non-deterministic as it uses `k8s.io/client-go/tools/cache.threadSafeMap` as the underlying cache.

Reply:
Client List order is deterministic. The interface this is a fake for is not the cache lister, it's the client that does live API lookups.

Reply:
Ohh, I see the difference, thanks for explaining. 
I checked Client List returns in alphabetical order, I can make do it in filterByNamespace, does it make sense to you?

Reply:
I have made the result deterministic, PTAL. Thanks.